tests/HUBViewControllerTests.m

```diff
@@ -1762,12 +1777,36 @@ - (void)testVisibleBodyComponents
     UICollectionViewCell * const cellC = [collectionViewDataSource collectionView:self.collectionView cellForItemAtIndexPath:indexPathC];
     
     self.collectionView.mockedVisibleCells = @[cellA, cellB, cellC];
-    
-    NSDictionary<NSIndexPath *, UIView *> * const visibleViews = self.viewController.visibleBodyComponentViews;
-    XCTAssertEqual(visibleViews.count, (NSUInteger)3);
-    XCTAssertEqual(visibleViews[[NSIndexPath indexPathWithIndex:0]], componentA.view);
-    XCTAssertEqual(visibleViews[[NSIndexPath indexPathWithIndex:1]], componentB.view);
-    XCTAssertEqual(visibleViews[[NSIndexPath indexPathWithIndex:2]], componentC.view);
+
+    NSDictionary<NSIndexPath *, UIView *> * const visibleHeaderViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeHeader];
+    XCTAssertEqual(visibleHeaderViews.count, (NSUInteger)1);
+    XCTAssertEqual(visibleHeaderViews[[NSIndexPath indexPathWithIndex:0]], headerComponent.view);
+
+    NSDictionary<NSIndexPath *, UIView *> * const visibleBodyViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeBody];
+    XCTAssertEqual(visibleBodyViews.count, (NSUInteger)3);
+    XCTAssertEqual(visibleBodyViews[[NSIndexPath indexPathWithIndex:0]], componentA.view);
+    XCTAssertEqual(visibleBodyViews[[NSIndexPath indexPathWithIndex:1]], componentB.view);
+    XCTAssertEqual(visibleBodyViews[[NSIndexPath indexPathWithIndex:2]], componentC.view);
+
+    NSDictionary<NSIndexPath *, UIView *> * const visibleOverlayViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeOverlay];
+    XCTAssertEqual(visibleOverlayViews.count, (NSUInteger)2);
+    XCTAssertEqual(visibleOverlayViews[[NSIndexPath indexPathWithIndex:0]], component1.view);
+    XCTAssertEqual(visibleOverlayViews[[NSIndexPath indexPathWithIndex:1]], component2.view);
+}
+
+- (void)testNoVisibleComponents
+{
+    [self simulateViewControllerLayoutCycle];
+
+    self.collectionView.mockedVisibleCells = @[];
+
+    NSDictionary<NSIndexPath *, UIView *> * const visibleHeaderViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeHeader];
+    NSDictionary<NSIndexPath *, UIView *> * const visibleBodyViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeBody];
+    NSDictionary<NSIndexPath *, UIView *> * const visibleOverlayViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeOverlay];
+
+    XCTAssertEqual(visibleHeaderViews.count, 0u);
```

Comment:
Oooooh! I didn't know you could do `0u`! That's great - then we can get rid of a lot of casting in the tests! 👍